packages/terra-date-picker/src/DateUtil.js

```diff
@@ -30,6 +30,42 @@ class DateUtil {
     return momentDates.length > 0 ? momentDates : dates;
   }
 
+  // Checks if a given date is out of the range between the start and end dates.
+  static isDateOutOfRange(sourceDate, startDate, endDate) {
+    if (!sourceDate || !sourceDate.isValid() ||
+      !startDate || !startDate.isValid() ||
+      !endDate || !endDate.isValid()) {
+      return false;
+    }
+
+    if (sourceDate.isBefore(startDate) && sourceDate.isAfter(endDate)) {
```

Comment:
this should be an ||

the sourceDate can't both be before the start and after the end dates

Reply:
Updated: https://github.com/cerner/terra-core/commit/f120301502e3d390e72a669fce130604de572ae5